common/network/uri_parser.c

```diff
@@ -32,24 +32,45 @@ bool uri_parse(char const *const uri, char *const scheme,
     scheme[length] = '\0';
   }
   offset += length;
-  if (strncmp(uri + offset, "://", 3) != 0) {
+  if (strncmp(uri + offset, "://[", 4) == 0) {
+    // IPv6
+    offset += 4;
+    if ((delim = strstr(uri + offset, "]:")) == NULL) {
+      return false;
+    }
+    length = (size_t)(delim - (uri + offset));
+    if (length > MAX_HOST_LENGTH || length + 1 > host_len) {
+      return false;
+    }
+    if (host != NULL) {
+      memcpy(host, uri + offset, length);
+      host[length] = '\0';
+    }
+    offset += length + 2;
+    if (port != NULL) {
+      *port = atoi(uri + offset);
+    }
+    return true;
+  } else if (strncmp(uri + offset, "://", 3) == 0) {
+    // IPv4
+    offset += 3;
+    if ((delim = strchr(uri + offset, ':')) == NULL) {
+      return false;
+    }
+    length = (size_t)(delim - (uri + offset));
+    if (length > MAX_HOST_LENGTH || length + 1 > host_len) {
+      return false;
+    }
+    if (host != NULL) {
+      memcpy(host, uri + offset, length);
+      host[length] = '\0';
+    }
+    offset += length + 1;
+    if (port != NULL) {
+      *port = atoi(uri + offset);
+    }
+    return true;
+  } else {
     return false;
```

Comment:
No need for an `else`, just `return false`, otherwise you don't have a return for the function